ttnn/cpp/ttnn/operations/conv/conv2d/device/optimized_conv_op_program_factory.cpp

```diff
@@ -162,8 +165,13 @@ std::vector<Tensor> OptimizedConvNew::create_output_tensors(const std::vector<Te
             uint32_t total_height_tiles = tt::tt_metal::compute_volume(output_shape) / output_shape[-1] / TILE_HEIGHT;
             uint32_t num_cores = total_height_tiles / this->parallelization_config.per_core_out_matrix_height_ntiles;
             CoreRangeSet shard_grid = tt::tt_metal::num_cores_to_corerange_set(num_cores, this->parallelization_config.grid_size, true);
-
             std::array<uint32_t, 2> shard_shape = {this->parallelization_config.per_core_out_matrix_height_ntiles * TILE_HEIGHT, output_shape[-1]};
+            if(this->use_max_cores){
```

Comment:
If you are doing work based on a boolean known ahead of time, it's usually better to have if/else statements rather than do all of the work for one case, and then if it's the other case do all of the work for the other case.

it'd be better to have
```
var1;
var2;
var3;
if (this->use_max_cores) {
  var1 = ...;
  var2 = ...;
  var3 = ...;
} else {
  var1 = ...;
  var2 = ...;
  var3 = ...;
}
```

Probably want to change it in some instances earlier as well.